uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerHost.java

```diff
@@ -293,15 +327,111 @@ private final String basicButtonHandler(@NonNull final String resourceButtonId,
         }
     }
 
-    @Override
-    public void setFlights(@Nullable final String flightsJson) {
-        Logger.i(TAG, "Set Flights..");
+    /**
+     * Gets all the accounts added
+     */
+    public List<String> getAllAccounts(final boolean expectingMultipleAccounts) {
         launch();
-        // Make the set flights UI visible on screen
-        UiAutomatorUtils.obtainChildInScrollable("Set Flights");
-        // input flights string in flights input box
-        UiAutomatorUtils.handleInput("com.microsoft.identity.testuserapp:id/editTextFlights", flightsJson);
-        // Click Set Flights button
-        UiAutomatorUtils.handleButtonClick("com.microsoft.identity.testuserapp:id/setFlightsButton");
+
+        final String resourceButtonId = "com.microsoft.identity.testuserapp:id/buttonGetAccounts";
+        final String textId = "AccountName";
+        if (shouldHandleFirstRun) {
+            handleFirstRun(); // handle first run experience
+        }
+        UiAutomatorUtils.handleButtonClick(resourceButtonId);
+        // Look for the dialog box
+        final UiObject dialogBox = UiAutomatorUtils.obtainUiObjectWithResourceId(
+                "android:id/message"
+        );
+        Assert.assertTrue(dialogBox.exists());
+
+        // As of now we are testing only for 2 accounts.
+        List<String> accounts = new ArrayList<>();
+        final String accountName = getDialogBoxText(dialogBox, textId);
+        if (accountName != null) {
+            accounts.add(accountName);
+        }
+        if (expectingMultipleAccounts) {
+            final String accountName2 = getDialogBoxText(dialogBox, textId);
+            if (accountName2 != null)
+                accounts.add(accountName2);
+        }
+        return accounts;
+    }
+
+    /**
+     * Removes the added account
+     */
+    public void removeAccount(@NonNull final String username) {
+        final UiObject removeAccount = UiAutomatorUtils.obtainUiObjectWithResourceId(
+                "com.microsoft.identity.testuserapp:id/buttonRemoveAccount"
+        );
+        UiAutomatorUtils.obtainChildInScrollable("someone@contoso.com");
+        UiAutomatorUtils.handleInput("com.microsoft.identity.testuserapp:id/editTextUsername", username);
+        try {
+            removeAccount.click();
+        } catch (final UiObjectNotFoundException e) {
+            throw new AssertionError(e);
+        }
+    }
+
+    /**
+     * Decode the sso token and look for the requested nonce
+     */
+    public void confirmNoncePresent() {
+        // fill the nonce
+        UiAutomatorUtils.obtainChildInScrollable("nonce");
```

Comment:
I'm not sure what we're doing here. It seems we're trying to obtain a UiObject and then forgetting about it. I don't think we even need to obtain this object though

Reply:
handled this in commit 060dcf041f3b0ff63dd5271687503dda4e4efc11